sklearn/cluster/tests/test_k_means.py

```diff
@@ -771,4 +774,85 @@ def test_x_squared_norms_init_centroids():
 def test_max_iter_error():
 
     km = KMeans(max_iter=-1)
-    assert_raise_message(ValueError, 'Number of iterations should be', km.fit, X)
+    assert_raise_message(ValueError, 'Number of iterations should be',
+                         km.fit, X)
+
+
+def test_kmeans_float32_64():
+    km = KMeans(n_init=1, random_state=11)
+
+    inertia = {}
+    X_new = {}
+
+    for dtype in [np.float64, np.float32]:
+        X_test = dtype(X)
+        km.fit(X_test)
+        # dtype of cluster centers has to be the dtype of the input data
+        assert_equal(km.cluster_centers_.dtype, dtype)
+        inertia[dtype] = km.inertia_
+        X_new[dtype] = km.transform(km.cluster_centers_)
```

Comment:
Did you mean `X_new[dtype] = km.transform(X)`?

`km.transform(km.cluster_centers_)` just returns an array of zeros...